--- a/frameworks/native/libs/binder/rust/Android.bp
+++ b/frameworks/native/libs/binder/rust/Android.bp
@@ -19,6 +19,7 @@
     host_supported: true,
     vendor_available: true,
     product_available: true,
+    recovery_available: true,
     target: {
         darwin: {
             enabled: false,
@@ -71,6 +72,7 @@
     host_supported: true,
     vendor_available: true,
     product_available: true,
+    recovery_available: true,
     target: {
         darwin: {
             enabled: false,
@@ -92,6 +94,7 @@
 rust_bindgen {
     name: "libbinder_ndk_bindgen",
     crate_name: "binder_ndk_bindgen",
+    recovery_available: true,
     wrapper_src: "sys/BinderBindings.hpp",
     source_stem: "bindings",
     bindgen_flag_files: [
