--- a/system/tools/aidl/build/aidl_interface_backends.go
+++ b/system/tools/aidl/build/aidl_interface_backends.go
@@ -391,15 +391,16 @@
 			"darwin": proptools.BoolPtr(false),
 			"":       nil,
 		}),
-		Crate_name:        rustCrateName,
-		Stem:              proptools.StringPtr("lib" + versionedRustName),
-		Defaults:          []string{"aidl-rust-module-defaults"},
-		Host_supported:    i.properties.Host_supported,
-		Vendor_available:  i.properties.Vendor_available,
-		Product_available: i.properties.Product_available,
-		Apex_available:    i.properties.Backend.Rust.Apex_available,
-		Min_sdk_version:   i.minSdkVersion(langRust),
-		Rustlibs:          i.properties.Backend.Rust.Additional_rustlibs,
+		Crate_name:         rustCrateName,
+		Stem:               proptools.StringPtr("lib" + versionedRustName),
+		Defaults:           []string{"aidl-rust-module-defaults"},
+		Host_supported:     i.properties.Host_supported,
+		Vendor_available:   i.properties.Vendor_available,
+		Product_available:  i.properties.Product_available,
+		Recovery_available: i.properties.Recovery_available,
+		Apex_available:     i.properties.Backend.Rust.Apex_available,
+		Min_sdk_version:    i.minSdkVersion(langRust),
+		Rustlibs:           i.properties.Backend.Rust.Additional_rustlibs,
 	}, &rust.SourceProviderProperties{
 		Source_stem: proptools.StringPtr(versionedRustName),
 	}, &aidlRustSourceProviderProperties{
