--- a/system/tools/aidl/build/aidl_interface_backends.go
+++ b/system/tools/aidl/build/aidl_interface_backends.go
@@ -399,15 +399,17 @@
 
 	versionedRustName := fixRustName(i.versionedName(version))
 	rustCrateName := fixRustName(i.ModuleBase.Name())
+	recoveryAvailable := i.properties.Recovery_available
 
 	mctx.CreateModule(wrapLibraryFactory(aidlRustLibraryFactory), &rustProperties{
-		Name:              proptools.StringPtr(rustModuleGen),
-		Crate_name:        rustCrateName,
-		Stem:              proptools.StringPtr("lib" + versionedRustName),
-		Defaults:          []string{"aidl-rust-module-defaults"},
-		Host_supported:    i.properties.Host_supported,
-		Vendor_available:  i.properties.Vendor_available,
-		Product_available: i.properties.Product_available,
+		Name:               proptools.StringPtr(rustModuleGen),
+		Crate_name:         rustCrateName,
+		Stem:               proptools.StringPtr("lib" + versionedRustName),
+		Defaults:           []string{"aidl-rust-module-defaults"},
+		Host_supported:     i.properties.Host_supported,
+		Vendor_available:   i.properties.Vendor_available,
+		Product_available:  i.properties.Product_available,
+		Recovery_available: recoveryAvailable,
 		Apex_available:    i.properties.Backend.Rust.Apex_available,
 		Min_sdk_version:   i.minSdkVersion(langRust),
 		Target:            rustTargetProperties{Darwin: darwinProperties{Enabled: proptools.BoolPtr(false)}},
