--- a/system/vold/KeystoreInfo.cpp
+++ b/system/vold/KeystoreInfo.cpp
@@ -57,8 +57,10 @@
 	}
 	std::string sql = "SELECT * FROM locksettings WHERE name = 'sp-handle' AND user = " + std::to_string(user_id);
 	rc = sqlite3_prepare_v2(db, sql.c_str(), -1, &stmt, NULL);
-	if (rc != SQLITE_OK)
+	if (rc != SQLITE_OK) {
+		sqlite3_close(db);
 		return "";
+	}
 	int64_t value = 0;
 	while ((rc = sqlite3_step(stmt)) == SQLITE_ROW) {
 		value = sqlite3_column_int64(stmt, 3);
